plugin/advisor/pg/advisor_naming_foreign_key_convention.go

```diff
@@ -0,0 +1,175 @@
+package pg
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*NamingFKConventionAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(advisor.Postgres, advisor.PostgreSQLNamingFKConvention, &NamingFKConventionAdvisor{})
+}
+
+// NamingFKConventionAdvisor is the advisor checking for foreign key naming convention.
+type NamingFKConventionAdvisor struct {
+}
+
+// Check checks for foreign key naming convention.
+func (check *NamingFKConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySQLReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, maxLength, err := advisor.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	checker := &namingFKConventionChecker{
+		level:        level,
+		title:        string(ctx.Rule.Type),
+		format:       format,
+		maxLength:    maxLength,
+		templateList: templateList,
+	}
+
+	for _, stmtNode := range root {
+		ast.Walk(checker, stmtNode)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return checker.adviceList, nil
+}
+
+type namingFKConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	title        string
+	format       string
+	maxLength    int
+	templateList []string
+}
+
+type indexMetaData struct {
+	indexName string
+	tableName string
+	metaData  map[string]string
+}
+
+// Visit implements ast.Visitor interface
+func (checker *namingFKConventionChecker) Visit(in ast.Node) ast.Visitor {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.Internal,
+				Title:   "Internal error for foreign key naming convention rule",
+				Content: fmt.Sprintf("%q meet internal error %q", in.Text(), err.Error()),
+			})
+			continue
+		}
+		if !regex.MatchString(indexData.indexName) {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.NamingFKConventionMismatch,
+				Title:   checker.title,
+				Content: fmt.Sprintf("Foreign key in table `%s` mismatches the naming convention, expect %q but found `%s`", indexData.tableName, regex, indexData.indexName),
```

Comment:
```suggestion
				Content: fmt.Sprintf("Foreign key in table \"%s\" mismatches the naming convention, expect %q but found \"%s\"", indexData.tableName, regex, indexData.indexName),
```
The \` is the mark for MySQL identifier, and for PG identifier, we should use `"`.